tests/rest/client/test_rooms.py

```diff
@@ -709,7 +709,7 @@ def test_post_room_no_keys(self) -> None:
         self.assertEqual(200, channel.code, channel.result)
         self.assertTrue("room_id" in channel.json_body)
         assert channel.resource_usage is not None
-        self.assertEqual(32, channel.resource_usage.db_txn_count)
+        self.assertEqual(36, channel.resource_usage.db_txn_count)
```

Comment:
I'm a bit surprised these transaction counts increase. AFAICT the only reason we would end up doing more transactions is if we start doing an `insert_delta_group_txn` but abort that because the delta chain is too long. Any ideas?

Reply:
I think its where we add the `replaces_state` to the unsigned section:

https://github.com/matrix-org/synapse/pull/13274/files#diff-d05c474c9fe45057f52616f38e54f6cdb3fa80a5a596ecb9c8fec3026ff8d68eR343-R346

So in the test we pull out some partial state for the group there, and then when we come to create the next event we pull out some other partial state for the group